pkg/quota/v1/evaluator/core/pods.go

```diff
@@ -394,8 +394,7 @@ func QuotaV1Pod(pod *corev1.Pod, clock clock.Clock) bool {
 	if pod.DeletionTimestamp != nil && pod.DeletionGracePeriodSeconds != nil {
 		now := clock.Now()
 		deletionTime := pod.DeletionTimestamp.Time
-		gracePeriod := time.Duration(*pod.DeletionGracePeriodSeconds) * time.Second
-		if now.After(deletionTime.Add(gracePeriod)) {
+		if *pod.DeletionGracePeriodSeconds != 0 && now.After(deletionTime) {
```

Comment:
looking at:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/rest/delete.go#L132

we can see that DeletionTimestamp is set based on current time + grace period as noted, so I agree it makes sense to not add the additional grace period to existing timestamp.

Given that, can't this check be simplified to just now.After(deletionTime) ?

Reply:
i think actually it is not just for simplification, it is incorrect to have the condition *pod.DeletionGracePeriodSeconds != 0

Reply:
Thanks for your review comments! @derekwaynecarr @yliaog  
Code updated.